lib/js/src/AesonSpec.js

```diff
@@ -163,14 +163,34 @@ function goldenSpec(decode, encode, name_of_type, json_file) {
 }
 
 function sampleGoldenSpec(decode, encode, name_of_type, json_file) {
-  AesonSpec_Jest.describe("AesonSpec.sampleGoldenSpec: " + (name_of_type + (" from file '" + (json_file + "'"))), (function (param) {
+  AesonSpec_Jest.describe("AesonSpec.sampleGoldenSpec: " + (name_of_type + (" from file '" + (json_file + "' with encoding utf8"))), (function (param) {
           var json = JSON.parse(Fs.readFileSync(json_file, "utf8"));
           AesonSpec_Jest.test("decode then encode json_file", (function (param) {
                   return sampleJsonRoundtripSpec(decode, encode, json);
                 }));
         }));
 }
 
+function encodingToString(encoding) {
+  if (encoding === "utf8") {
+    return "utf8";
+  } else if (encoding === "base64") {
+    return "base64";
+  } else if (encoding === "latin1") {
+    return "latin1";
+  } else if (encoding === "binary") {
+    return "binary";
+  } else if (encoding === "hex") {
+    return "hex";
+  } else if (encoding === "ascii") {
+    return "ascii";
+  } else if (encoding === "utf16le") {
+    return "utf16le";
+  } else {
+    return "ucs2";
+  }
+}
+
```

Comment:
This is seems redundant in the generated code but makes more sense in the original .ml file